pandas/tests/io/test_sql.py

```diff
@@ -564,6 +564,13 @@ class DummyException(Exception):
         res2 = self.pandasSQL.read_query("SELECT * FROM test_trans")
         assert len(res2) == 1
 
+    def _to_sql_with_negative_npinf(self):
```

Comment:
Don't think this is the right spot for this test. Think this needs to go in `_TestSQLAlchemy` so it hits different database flavors.

Also we usually test round trip behavior. Create DataFrame, to_sql, read_sql, `tm.assert_frame_equal(result, expected)`

Reply:
@mroeschke Thanks for the feeback!

I've moved the test to `_TestSQLAlchemy` like you suggested and added the round trips. I also fixed the `whatsnew` entry